jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1784,7 +1784,7 @@ public boolean equalsIgnoreCase(String string) {
 	 */
 	@Deprecated(forRemoval=false, since="1.1")
 	public void getBytes(int start, int end, byte[] data, int index) {
-		if (0 <= start && start <= end && end <= lengthInternal()) {
+		if (0 <= start && start <= end && end <= lengthInternal() && 0 <= index && ((end - start) <= (data.length - index))) {
```

Comment:
Can we apply the same fix to the `char[] data` overload right below (for both Java 8 and Java 9+)?

Reply:
The `char[]` overload of `getBytes` is package protected, and is used from StringBuilder / StringBuffer. The parameters can be assumed valid. For performance reasons we should avoid adding the checks there.

Reply:
Right, thanks.